client/app/scripts/superdesk-archive/directives.js

```diff
@@ -349,6 +349,24 @@
                     scope.toggleSelected = function(item) {
                         multi.toggle(item);
                     };
+
+                    /**
+                     * Get actions type based on item state. Used with activity filter.
+                     *
+                     * @param {Object} item
+                     * @returns {string}
+                     */
+                    scope.getType = function(item) {
+                        if (item.state === 'spiked') {
+                            return 'spike';
+                        } else if (item._type === 'published' && item.allow_post_publish_actions === true) {
+                            return 'archive';
+                        } else if (item._type === 'published' && item.allow_post_publish_actions === false) {
+                            return 'archived';
+                        }
```

Comment:
Just an idea - do we have some kind of an enum-ish object containing all possible item types defined in the client application as well?